src/viam/sdk/CMakeLists.txt

```diff
@@ -276,10 +276,10 @@ target_link_libraries(viamsdk
   PRIVATE Threads::Threads
 )
 
-# TODO(RSDK-10366): Currently, rust_utils is not published for
-# windows, so don't link to it.  Instead, link a stub implementation
-# that just calls `abort`.
-if (NOT WIN32)
+# rust-utils doesn't build in CI on windows other than x86_64, so confirm that we're either not
+# windows, or if we are windows that we're AMD64, or else that the user has provided their own
+# build of rust_utils.
+if (NOT WIN32 OR num_viam_rust_utils_files GREATER 0 OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "AMD64")
```

Comment:
Could this test just look only at `num_viam_rust_utils_files` now? I'd assume that if it is greater than zero then one was found and it should be used.

Reply:
I may be wrong, but is this a correct assumption? I thought `num_viam_rust_utils_files` was evaluated _before_ we downloaded, in order to determine if we _should_ download a `rust_utils` file. So the `not WIN32 or AMD64` cases are important for the cases where the user hasn't provided a `rust-utils` file but we can download one for them.

Reply:
Sorry, you are right, I got myself tangled up. I was reacting to having complex logic to decide whether to use rust utils or build the stub here. It just feels like it would be better if all of that tortured logic was isolated up in the top level CMakeLists.txt and this file had a way of making a simple yes/no decision about whether to use the library or build the stubs.

Actually, I wonder if `if (TARGET...` is what I'm after: https://cmake.org/cmake/help/latest/command/if.html#target

Again, I know you are eager to land this, but could you please add a TODO (doesn't need a ticket, leave it for someone to clean up later).

Reply:
went ahead and switched to `if (TARGET viam_rust_utils)`, confirmed that it builds correctly both with a user-provided `rust-utils` file and without one.